### DISCUSSION TITLE
Best way to test with uncommitted changes?

### DISCUSSION THREAD

**Discussion by macintacos** [copier-org/copier#743] [Q&A]
I'm working on developing a template, and want to be able to test the template without necessarily needing to commit my changes (since they'll be changing pretty frequently as I prototype). What's the best way to go about doing that?

**Comment by mshafer-NI** [comment]
What I and my team-mates have done is:
* Make a dev branch
* Go ahead and commit as we work to have copier use the latest committed
* When we think the template is ready - use Git to merge the dev commits (I tend to use `rebase -i main`) OR, go ahead and post a PR and have it merge with a squash commit.

**Comment by yajo** [comment]
Just use copier to copy the local template. Since
https://github.com/copier-org/copier/pull/520, dirty changes should be
included.

**Reply by mshafer-NI** [reply]
Is this in 6.1.0?

**Reply by macintacos** [reply]
I'm on 6.1.0, but I've been running with `--vcs-ref=HEAD` after doing a WIP commit, which seems to be what this PR introduced. Should I not be including that flag anymore?

**Reply by macintacos** [reply]
When I remove the `--vcs-ref=HEAD` flag and run it manually after making some minor change (like changing a "help" prompt), I do see the message that says the following, which makes me think that it _should_ be working:

```
/Users/macinburrito/Library/Caches/pypoetry/virtualenvs/ops-template-t_ulhZ_c-py3.10/lib/python3.10/site-packages/copier/vcs.py:151: DirtyLocalWarning: Dirty template changes included automatically.
  warn(
```

However, when it shows the prompt that I changed in the `copier.yml` file I don't see that changes reflected. The only way I can get it to work is if I commit the WIP change manually and add `--vcs-ref=HEAD`, which seems to be what should already be handled by that PR. Any insight into what I'm doing wrong?

**Reply by yajo** [reply]
You shouldn't need to commit. Copier should do the commit in another temporary worktree for you.

If you've spotted a bug in that feature, then you will have to provide valid steps to reproduce it. Otherwise I can't fix it 😕

**Reply by FlorianLudwig** [reply]
I have created a ticket for this bug with a description on how to reproduce it here: https://github.com/copier-org/copier/issues/879
